src/ServiceControl/Operations/ErrorIngestion.cs

```diff
@@ -0,0 +1,86 @@
+﻿namespace ServiceControl.Operations
+{
+    using System;
+    using System.Threading;
+    using System.Threading.Tasks;
+    using NServiceBus;
+    using NServiceBus.Raw;
+    using Raven.Client;
+    using Recoverability;
+    using ServiceBus.Management.Infrastructure.Settings;
+
+    class ErrorIngestion
+    {
+        public ErrorIngestion(ErrorIngestor errorIngestor, string errorQueue, RawEndpointFactory rawEndpointFactory, IDocumentStore documentStore, LoggingSettings loggingSettings, Func<string, Exception, Task> onCriticalError)
+        {
+            this.errorIngestor = errorIngestor;
+            this.errorQueue = errorQueue;
+            this.rawEndpointFactory = rawEndpointFactory;
+            this.onCriticalError = onCriticalError;
+            importFailuresHandler = new SatelliteImportFailuresHandler(documentStore, loggingSettings, onCriticalError);
+        }
+
+        public async Task EnsureStarted()
```

Comment:
Why not flow the cancellation token from watchdog into this method so that the waitasync could be cancelled if needed and we avoid indefinite hanging?